### DISCUSSION TITLE
Fuseki system identifiers for blank nodes - RDF Patch

### DISCUSSION THREAD

**Discussion by edmondchuc** [apache/jena#2198] [Q&A] [answered]
Hi, I'm trying to figure out whether Fuseki provides a way to retrieve the system identifiers for blank nodes like in [GraphDB](https://graphdb.ontotext.com/documentation/10.0/query-behaviour.html#how-to-access-internal-identifiers-for-entities).

The reason why I want the system identifiers is to be able to generate an RDF Patch log entry to update the values for blank node objects in a Fuseki system. The [docs for RDF Patch](https://afs.github.io/rdf-delta/rdf-patch.html#blank-nodes) describe updating values within blank nodes by referring to the system's identifier. 

Is my use case supported? Or is there another way to achieve the same result without using SPARQL Update? Thanks.

**Comment by afs** [comment, marked as answer]
In Fuseki `str(?myBlankNode)` will return the internal identifier as a string.
It's not a portable - it is  an extension.

Otherwise use SPARQL Update - that is portable.

**Reply by edmondchuc** [reply]
Perfect, thanks for the quick response @afs!
